plugins/processors/dedup/dedup.go

```diff
@@ -0,0 +1,89 @@
+package dedup
+
+import (
+	"log"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+var sampleConfig = `
+  ## Maximum time to suppress output
+  dedup_interval = "600s"
+  ## Maximum time to keep cached metric without update
+  evict_interval = "1h"
+`
+
+type Dedup struct {
+	DedupInterval internal.Duration `toml:"dedup_interval"`
+	EvictInterval internal.Duration `toml:"evict_interval"`
+	Cache         map[uint64]telegraf.Metric
+}
+
+func NewDedup() *Dedup {
+	return &Dedup{
+		DedupInterval: internal.Duration{Duration: 10 * time.Minute},
+		EvictInterval: internal.Duration{Duration: 1 * time.Hour},
+		Cache:         make(map[uint64]telegraf.Metric),
+	}
+}
+
+func (d *Dedup) SampleConfig() string {
+	return sampleConfig
+}
+
+func (d *Dedup) Description() string {
+	return "Deduplicate repetitive metrics"
+}
+
+// Remove single item from slice
+func remove(slice []telegraf.Metric, i int) []telegraf.Metric {
+	slice[len(slice)-1], slice[i] = slice[i], slice[len(slice)-1]
+	return slice[:len(slice)-1]
+}
+
+// Remove expired items from cache
+func (d *Dedup) cleanup() {
+	keep := make(map[uint64]telegraf.Metric, 0)
```

Comment:
Since the processors get called for every metric, I think we should limit how often this is called and reuse the cache map.